### PR TITLE
Allow for authentication for configured proxy

### PR DESCRIPTION
**Pull request by marloncarvalho**
This PR fixes issue #44. @lpatino10

### CODE REVIEW

File: src/test/java/com/ibm/cloud/sdk/core/test/http/HttpConfigTest.java [hunk 1]
```diff
@@ -16,13 +22,23 @@
...
+      public Request authenticate(@Nullable Route route, Response response) throws IOException {
+        return null;
+      }
+    };
+
     Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("proxyHost", 8080));
     HttpConfigOptions configOptions = new HttpConfigOptions.Builder()
         .disableSslVerification(true)
         .proxy(proxy)
+            .proxyAuthenticator(authenticator)
```
Comment (lpatino10) [inline src/test/java/com/ibm/cloud/sdk/core/test/http/HttpConfigTest.java]: Could you align the indentation here?

Comment (marloncarvalho) [reply src/test/java/com/ibm/cloud/sdk/core/test/http/HttpConfigTest.java]: Wow! What a silly mistake! Sorry. I've fixed it. :)

**Comment by germanattanasio** [comment]
@marloncarvalho you also need to configure git because your user doesn't match your commit

**Comment by marloncarvalho** [comment]
I see! It's using my enterprise username and e-mail. Sorry. I've changed it. What should I do now? Just commit something and push it? Fork again?

**Comment by marloncarvalho** [comment]
Nevermind! I didn't know I can change the Author with rebase and amend. Fixed. @lpatino10 @germanattanasio

**Pull request merged**